lib/ansible/modules/remote_management/manageiq/manageiq_alert_profiles.py

```diff
@@ -0,0 +1,319 @@
+              'ExtManagementSystem', 'MiddlewareServer']
+  alerts:
+    description:
+      - List of alert descriptions to assign to this profile.
+      - Required if state is "present"
+  notes:
+    description:
+      - Optional notes for this profile
+
+'''
+
+EXAMPLES = '''
+- name: List alert profiles in ManageIQ
+  manageiq_alert_profiles:
+    state: list
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Add an alert profile to ManageIQ
+  manageiq_alert_profiles:
+    state: present
+    name: Test profile
+    resource_type: ContainerNode
+    alerts:
+      - Test Alert 01
+      - Test Alert 02
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete an alert profile from ManageIQ
+  manageiq_alert_profiles:
+    state: absent
+    name: Test profile
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+class ManageIQAlertProfiles(object):
+    """ Object to execute alert profile management operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+        self.url = '{api_url}/alert_definition_profiles'.format(api_url=self.api_url)
+
+    def get_profiles(self):
+        """ Get all alert profiles from ManageIQ
+        """
+        try:
+            response = self.client.get(self.url + '?expand=alert_definitions,resources')
+        except Exception as e:
+            self.module.fail_json(msg="Failed to query alert profiles: {error}".format(error=e))
+        return response.get('resources') or []
+
+    def get_alerts(self, alert_descriptions):
+        """ Get a list of alert hrefs from a list of alert descriptions
+        """
+        alerts = []
+        for alert_description in alert_descriptions:
+            alert = self.manageiq.find_collection_resource_or_fail("alert_definitions",
+                                                                   description=alert_description)
+            alerts.append(alert['href'])
+
+        return alerts
+
+    def add_profile(self, profile):
+        """ Add a new alert profile to ManageIQ
+        """
+        # find all alerts to add to the profile
+        # we do this first to fail early if one is missing.
+        alerts = self.get_alerts(profile['alerts'])
+
+        # build the profile dict to send to the server
+
+        profile_dict = dict(name=profile['name'],
+                            description=profile['name'],
```

Comment:
Shouldn't this honor description param, if supplied?

Reply:
Is there any usecase for that?

Having a description param might confuse users, because the ManageIQ UI is counter-intuative. People might assume "name" is the name visible in the UI, and the description is a space for longer description of what the alert does, while the UI actually shows the "description" where one would expect to find a "name".

Reply:
Oh, I see there is no description param in code.  That's OK with me.
Please remove it from DOCUMENTATION then :-)